### ISSUE TITLE
Auto-switch to device when it becomes available

### ISSUE THREAD

**Issue by vertigo220** [File-New-Project/EarTrumpet#519]
I switch between using my HDMI monitor and an SPDIF soundbar for playing audio from my computer. I'd like it to always use the soundbar when it's available, and switch back to the monitor when it's not. I'm not even sure if this would be possible, since the soundbar always shows as available even when it's on a different input, as optical clearly doesn't have a way of telling if the device is actually using that input.

**Comment by riverar** [comment]
We have something like this built and will be available as an add-on soon. Stay tuned.

**Comment by vertigo220** [comment]
I'm curious, if this works with SPDIF, how? (assuming it could even be explained briefly and non-technically)

**Comment by riverar** [comment]
@vertigo220 Thanks, I skimmed over that a bit too quickly.

It's not immediately clear to me why an optical sink can't signal back to a renderer, probably cost related, but I'm still reviewing the specs. In the near future at least, you could sense the sink's availability (e.g. a photodiode over a power led?) and communicate with the PC, then get picked up by a future EarTrumpet addon, and change the default device. It's one, admittedly complex, solution.

**Comment by vertigo220** [comment]
Yeah. I _think_ they're one-way, whether due to the output being transmit only and input being receive only, which would make sense since optical would require totally different elements for transmit and receive, unlike other connectors, or perhaps being a single conductor, they can't send signals both ways simultaneously. By the way, I mixed up terminology somewhat, and should have specified TOSLINK vs S/PDIF, since all TOSLINK are S/PDIF but not all S/PDIF are TOSLINK.

I don't think the diode over the power LED would work, since a) my soundbar doesn't have one, and b) it would only indicate it was on, not that it was switched to the optical input. The only thing I could think of would be to use a mic to listen for sound, and if it's using that output and something is playing but the mic isn't picking up the sound, then it would assume the device isn't switched to that input and switch to a secondary device. Of course, this has its own set of problems, and wouldn't work for determining when the optical device is available. Perhaps better would be to always play the audio to the optical device, and mute or don't play it on the secondary device when it's picked up from the optical one, but again, that would have a whole mess of issues.

Realistically, I doubt this is possible, and it's such a minor issue it's probably not worth the effort if it were. I just figured I'd check just in case.
